interfaces/Portalicious/src/app/pages/users/add-user-form/add-user-form.component.ts

```diff
@@ -0,0 +1,144 @@
+
+import { FormSidebarComponent } from '~/components/form/form-sidebar.component';
+import { FormFieldWrapperComponent } from '~/components/form-field-wrapper/form-field-wrapper.component';
+import { UserApiService } from '~/domains/user/user.api.service';
+import { User } from '~/domains/user/user.model';
+import { ToastService } from '~/services/toast.service';
+import {
+  generateFieldErrors,
+  genericFieldIsRequiredValidationMessage,
+} from '~/utils/form-validation';
+
+type AddUserToTeamFormGroup =
+  (typeof AddUserFormComponent)['prototype']['formGroup'];
+
+@Component({
+  selector: 'app-add-user-form',
+  styles: ``,
+  standalone: true,
+  templateUrl: './add-user-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    ButtonModule,
+    FormSidebarComponent,
+    FormFieldWrapperComponent,
+    DropdownModule,
+    MultiSelectModule,
+    InputTextModule,
+    ReactiveFormsModule,
+  ],
+})
+export class AddUserFormComponent {
+  userToEdit = input<undefined | User>();
+  formVisible = model.required<boolean>();
+
+  private userApiService = inject(UserApiService);
+  private toastService = inject(ToastService);
+
+  constructor() {
+    effect(
+      () => {
+        const user = this.userToEdit();
+        if (user) {
+          this.formGroup.patchValue({
+            usernameValue: user.displayName,
+            emailValue: user.username,
+          });
+          this.formGroup.controls.emailValue.disable();
+        } else {
+          this.formGroup.controls.emailValue.enable();
+        }
+      },
+      {
+        allowSignalWrites: true,
+      },
+    );
+  }
+
+  isEditing = computed(() => !!this.userToEdit());
+
+  allUsers = injectQuery(this.userApiService.getAllUsers());
+
+  formGroup = new FormGroup({
+    usernameValue: new FormControl<string>('', {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      validators: [Validators.required],
+      nonNullable: true,
+    }),
+    emailValue: new FormControl<string>('', {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      validators: [Validators.required, Validators.email],
+      nonNullable: true,
+    }),
+  });
+
+  formFieldErrors = generateFieldErrors<AddUserToTeamFormGroup>(
+    this.formGroup,
+    {
+      usernameValue: genericFieldIsRequiredValidationMessage,
+      emailValue: (control) => {
+        if (!control.invalid) {
+          return;
+        }
+        return $localize`Enter a valid email address`;
+      },
+    },
+  );
+
+  addUserMutation = injectMutation(() => ({
+    mutationFn: ({
+      usernameValue,
+      emailValue,
+    }: {
+      usernameValue: string;
+      emailValue: string;
+    }) => {
+      if (this.isEditing() && this.userToEdit()?.id) {
+        return this.userApiService.udpateUserDisplayName({
+          id: this.userToEdit()?.id,
+          displayName: usernameValue,
+        });
+      }
```

Comment:
```suggestion
      if (this.isEditing()) {
        return this.userApiService.udpateUserDisplayName({
          id: this.userToEdit()?.id,
          displayName: usernameValue,
        });
      }
```

We don't need the extra check on the id, because it's already happening in udpateUserDisplayName. As-is, there is an unlikely risk that we create user when we mean to edit it.